container/docker/factory.go

```diff
@@ -273,6 +269,9 @@ func Register(factory info.MachineInfoFactory, fsInfo fs.FsInfo) error {
 				return fmt.Errorf("cAdvisor requires docker version %v or above but we have found version %v reported as \"%v\"", expected_version, dockerVersion, version_string)
 			}
 		}
+		if dockerVersion[0] == 1 && dockerVersion[1] >= 10 {
```

Comment:
`sudo docker daemon --exec-opt="native.cgroupdriver=systemd" -s
devicemapper` will make the daemon use systemd. I wonder if we can use
docker info to figure this out.

On Fri, Feb 5, 2016 at 12:56 PM, Jimmi Dyson <notifications@github.com>
wrote:

> In container/docker/factory.go
> <https://github.com/google/cadvisor/pull/1095#discussion_r52071462>:
>
> > @@ -273,6 +269,9 @@ func Register(factory info.MachineInfoFactory, fsInfo fs.FsInfo) error {
> >  				return fmt.Errorf("cAdvisor requires docker version %v or above but we have found version %v reported as \"%v\"", expected_version, dockerVersion, version_string)
> >  			}
> >  		}
> > +		if dockerVersion[0] == 1 && dockerVersion[1] >= 10 {
>
> I run fedora & this works for me. Docker has changed cgroup hierarchy in
> 1.10 it seems.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/1095/files#r52071462>.
>